tests/test_transforms/test_transform_processing.py

```diff
@@ -0,0 +1,280 @@
+    with pytest.raises(TypeError):
+        data_str = '123'
+        _ = to_tensor(data_str)
+
+
+class TestToTensor:
+
+    def test_init(self):
+        TRANSFORM = ToTensor(keys=['img_label'])
+        assert TRANSFORM.keys == ['img_label']
+
+    def test_transform(self):
+        TRANSFORMS = ToTensor(['instances.bbox', 'img_label', 'bbox'])
+
+        # Test multi-level key and single-level key (multi-level key is
+        # not in results)
+        results = {'instances': {'label': [1]}, 'img_label': [1]}
+        results_tensor = TRANSFORMS.transform(copy.deepcopy(results))
+        assert isinstance(results_tensor['instances']['label'], list)
+        assert isinstance(results_tensor['img_label'], torch.Tensor)
+
+        # Test multi-level key (multi-level key is in results)
+        results = {'instances': {'bbox': [1]}, 'img_label': [1]}
+        results_tensor = TRANSFORMS.transform(copy.deepcopy(results))
+        assert isinstance(results_tensor['instances']['bbox'], torch.Tensor)
+
+    def test_repr(self):
+        TRANSFORMS = ToTensor(['instances.bbox', 'img_label'])
+        TRANSFORMS_str = str(TRANSFORMS)
+        isinstance(TRANSFORMS_str, str)
+
+
+class TestImageToTensor:
+
+    def test_init(self):
+        TRANSFORMS = ImageToTensor(['img'])
+        assert TRANSFORMS.keys == ['img']
+
+    def test_transform(self):
+        TRANSFORMS = ImageToTensor(['img'])
+
+        # image only has one channel
+        results = {'img': np.zeros((224, 224))}
+        results = TRANSFORMS.transform(results)
+        assert results['img'].shape == (1, 224, 224)
+
+        # image has three channels
+        results = {'img': np.zeros((224, 224, 3))}
+        results = TRANSFORMS.transform(results)
+        assert results['img'].shape == (3, 224, 224)
+
+    def test_repr(self):
+        TRANSFORMS = ImageToTensor(['img'])
+        TRANSFORMS_str = str(TRANSFORMS)
+        assert isinstance(TRANSFORMS_str, str)
+
+
+class TestRandomFlip:
+
+    def test_init(self):
+
+        # prob is float
+        TRANSFORMS = RandomFlip(0.1)
+        assert TRANSFORMS.prob == 0.1
+
+        # prob is None
+        TRANSFORMS = RandomFlip(None)
+        assert TRANSFORMS.prob is None
+
+        # prob is a list
+        TRANSFORMS = RandomFlip([0.1, 0.2], ['horizontal', 'vertical'])
+        assert len(TRANSFORMS.prob) == 2
+        assert len(TRANSFORMS.direction) == 2
+
+        # direction is an invalid type
+        with pytest.raises(ValueError):
+            TRANSFORMS = RandomFlip(0.1, 1)
+
+        # prob is an invalid type
+        with pytest.raises(ValueError):
+            TRANSFORMS = RandomFlip('0.1')
+
+    def test_transform(self):
+
+        results = {
+            'img': np.random.random((224, 224, 3)),
+            'gt_bboxes': np.array([[0, 1, 100, 101]]),
+            'gt_keypoints': np.array([[100, 100]]),
+            'gt_semantic_seg': np.random.random((224, 224, 3))
+        }
```

Comment:
Maybe `flip` and `flip_direction` can be added to `results`, so that we can see whether `override` works.